telegram/ext/_application.py

```diff
@@ -0,0 +1,1212 @@
+        # function
+        task = asyncio.create_task(
+            self.__create_task_callback(
+                coroutine=coroutine, update=update, is_error_handler=is_error_handler
+            )
+        )
+
+        if self.running:
+            self.__create_task_tasks.add(task)
+            task.add_done_callback(self.__create_task_done_callback)
+        else:
+            _logger.warning(
+                "Tasks created via `Application.create_task` while the application is not "
+                "running won't be automatically awaited!"
+            )
+
+        return task
+
+    def __create_task_done_callback(self, task: asyncio.Task) -> None:
+        # We just retrieve the eventual exception so that asyncio doesn't complain in case
+        # it's not retrieved somewhere else
+        task.exception()
+        self.__create_task_tasks.discard(task)
+
+    async def __create_task_callback(
+        self,
+        coroutine: Coroutine[Any, Any, _RT],
+        update: object = None,
+        is_error_handler: bool = False,
+    ) -> _RT:
+        try:
+            return await coroutine
+        except Exception as exception:
+            if isinstance(exception, ApplicationHandlerStop):
+                warn(
+                    'ApplicationHandlerStop is not supported with asynchronously '
+                    'running handlers.',
+                    stacklevel=1,
+                )
+
+            # Avoid infinite recursion of error handlers.
+            elif is_error_handler:
+                _logger.exception(
+                    'An error was raised and an uncaught error was raised while '
+                    'handling the error with an error_handler.',
+                    exc_info=exception,
+                )
+
+            else:
+                # If we arrive here, an exception happened in the task and was neither
+                # ApplicationHandlerStop nor raised by an error handler.
+                # So we can and must handle it
+                await self.dispatch_error(update, exception, coroutine=coroutine)
+
+            # Raise exception so that it can be set on the task
+            raise exception
+        finally:
+            self._mark_for_persistence_update(update=update)
+
+    async def _update_fetcher(self) -> None:
+        # Continuously fetch updates from the queue. Exit only once the signal object is found.
+        while True:
+            update = await self.update_queue.get()
+
+            if update is _STOP_SIGNAL:
+                _logger.debug('Dropping pending updates')
+                while not self.update_queue.empty():
+                    self.update_queue.task_done()
+
+                # For the _STOP_SIGNAL
+                self.update_queue.task_done()
+                return
+
+            _logger.debug('Processing update %s', update)
+
+            if self._concurrent_updates:
+                asyncio.create_task(self.__process_update_wrapper(update))
```

Comment:
we're deliberately not `await`ing this right? Because it has to run concurrently? Awaiting this would "block" it in a way

Reply:
Correct. Though it's good that you commented on it: I'll change to `self.create_task`, which makes sure that the task is awaited on shutdown :)